include/dr/mhp/global.hpp

```diff
@@ -145,6 +161,20 @@ inline sycl::queue select_queue(MPI_Comm comm = MPI_COMM_WORLD) {
       devices[dr::communicator(comm).rank() % rng::size(devices)]);
 }
 
+inline void init(sycl::queue q) {
+  __detail::initialize_mpi();
```

Comment:
My one reservation about this PR is this line.  According to the MPI spec, we technically don't have a guarantee that all the processes exist until *after* `MPI_Init` is called.  This means that `sycl::queue` might not be valid on all processes after we call `initialize_mpi` here.

However, for most MPI implementations, the processes *will* exist before `MPI_Init` is called, so they will each have their corresponding queues and this code will run fine.

The option would be to remove this line `__detail::initialize_mpi()` for `init(sycl::queue q)`, which would require users to always call `MPI_Init` themselves if they pass in a queue.